src/test/java/org/folio/circulation/resources/RequestQueueResourceTest.java

```diff
@@ -0,0 +1,315 @@
+    loansFixture.checkOutByBarcode(item, rebecca);
+
+    IndividualResource firstHoldRequest = holdRequest(steve);
+    IndividualResource secondHoldRequest = holdRequest(james);
+    IndividualResource firstRecallRequest = recallRequest(charlotte);
+    IndividualResource secondRecallRequest = recallRequest(jessica);
+
+    JsonObject initialReorder = new ReorderQueueBuilder()
+      .addReorderRequest(firstHoldRequest.getId().toString(), initialState[0])
+      .addReorderRequest(secondHoldRequest.getId().toString(), initialState[1])
+      .addReorderRequest(firstRecallRequest.getId().toString(), initialState[2])
+      .addReorderRequest(secondRecallRequest.getId().toString(), initialState[3])
+      .create();
+
+    JsonObject initialReorderResponse = requestQueueFixture
+      .reorderQueue(item.getId().toString(), initialReorder);
+
+    assertQueue(initialReorder, initialReorderResponse);
+
+    JsonObject subsequentReorder = new ReorderQueueBuilder()
+      .addReorderRequest(firstHoldRequest.getId().toString(), targetState[0])
+      .addReorderRequest(secondHoldRequest.getId().toString(), targetState[1])
+      .addReorderRequest(firstRecallRequest.getId().toString(), targetState[2])
+      .addReorderRequest(secondRecallRequest.getId().toString(), targetState[3])
+      .create();
+
+    JsonObject subsequentReorderResponse = requestQueueFixture
+      .reorderQueue(item.getId().toString(), subsequentReorder);
+
+    assertQueue(subsequentReorder, subsequentReorderResponse);
+  }
+
+  private IndividualResource pageRequest(IndividualResource requester) throws Exception {
+    return requestsFixture.place(
+      new RequestBuilder()
+        .open()
+        .page()
+        .forItem(item)
+        .by(requester)
+        .withPickupServicePoint(servicePointsFixture.cd1())
+    );
+  }
+
+  private IndividualResource holdRequest(IndividualResource requester) throws Exception {
+    return requestsFixture.place(
+      new RequestBuilder()
+        .open()
+        .hold()
+        .forItem(item)
+        .by(requester)
+        .withPickupServicePoint(servicePointsFixture.cd1())
+    );
+  }
+
+  private IndividualResource recallRequest(IndividualResource requester) throws Exception {
+    return requestsFixture.place(
+      new RequestBuilder()
+        .open()
+        .recall()
+        .forItem(item)
+        .by(requester)
+        .withPickupServicePoint(servicePointsFixture.cd1())
+    );
+  }
+
+  private IndividualResource inFulfillmentRecallRequest(IndividualResource requester) throws Exception {
+    return requestsFixture.place(
+      new RequestBuilder()
+        .open()
+        .recall()
+        .withStatus(RequestBuilder.OPEN_IN_TRANSIT)
+        .forItem(item)
+        .by(requester)
+        .withPickupServicePoint(servicePointsFixture.cd1())
+    );
+  }
+
+  private void assertQueue(JsonObject initialQueue, JsonObject reorderedQueue)
```

Comment:
Is this attempted to verify that the new state of the queue is what was requested from the reorder?

If so, an alternative could be to check the size of the queue and then a single loop to check that each request is present and in the matching position. This would avoid the need for sorting.

Why are there two loops? Is this because one checks that the queue provided in the response is correct, and the other is the response from a separate request to get the queue?

If so, could the common checking logic be extracted and this be simplified to two calls to the same checks.

Reply:
The first loop to verify that requested reorder and response are the same.
The second loop to verify that requested reorder and actual persisted queue positions are the same.

>> could the common checking logic be extracted and this be simplified to two calls to the same checks.

Sure

Reply:
fixed